scanner/src/main/java/ca/on/oicr/gsi/runscanner/scanner/processor/RevioPacBioProcessor.java

```diff
@@ -0,0 +1,399 @@
+    for (RevioPacBioProcessor.ProcessMetadata processor : REVIO_METADATA_PROCESSORS) {
+      try {
+        processor.accept(metadata, dto, timeZone);
+      } catch (XPathException e) {
+        log.error("Failed to extract metadata", e);
+      }
+    }
+  }
+
+  /**
+   * Tests whether a String is blank (empty or just spaces) or null. Duplicated from MISO's
+   * LimsUtils.
+   *
+   * @param s String to test for blank or null
+   * @return true if blank or null String provided
+   */
+  private static boolean isStringBlankOrNull(String s) {
+    return s == null || s.trim().isEmpty();
+  }
+
+  @Override
+  public PathType getPathType() {
+    return PathType.DIRECTORY;
+  }
+
+  /**
+   * Returns a stream that includes Metadata directory level to be used for further processing
+   *
+   * @param runDirectory which we are currently processing
+   * @param directoryName specific sub-directory we want
+   * @return Stream from Metadata directory level
+   */
+  private Stream<File> streamSmrtCellSubdirectories(File runDirectory, String directoryName) {
+    return Stream.of(runDirectory.listFiles())
+        .filter(
+            cellDirectory ->
+                cellDirectory.isDirectory() && REVIO_CELL_DIRECTORY.test(cellDirectory.getName()))
+        .flatMap(
+            cellDirectory ->
+                Stream.of(cellDirectory.listFiles())
+                    .filter(
+                        subDirectory ->
+                            subDirectory.isDirectory()
+                                && subDirectory.getName().equals(directoryName)));
+  }
+
+  /**
+   * Returns the time a file was created
+   *
+   * @param file which we are currently processing
+   * @return Instant point in time
+   */
+  private static Instant getFileCreationTime(File file) {
+    try {
+      FileTime fileTime =
+          (FileTime) Files.getAttribute(Path.of(file.getAbsolutePath()), "creationTime");
+      return fileTime.toInstant();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * Grab the completion time from a log file
+   *
+   * @param file we are checking
+   * @return completion time
+   */
+  private static Instant getLogCompletionTime(File file) {
+    try {
+      Scanner myReader = new Scanner(file);
+      Pattern pattern =
+          Pattern.compile("^\\[INFO] (.*) \\[.*] exiting with return code \\d+" + " .*");
+      while (myReader.hasNextLine()) {
+        Matcher matcher = pattern.matcher(myReader.nextLine());
+        if (matcher.matches()) {
+          String stringDate = matcher.group(1);
+          String datePattern = "yyyy-MM-dd HH:mm:ss[.SSS][,SSS][Z][X]";
+          DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern(datePattern);
+          return LocalDateTime.parse(stringDate, dateTimeFormatter)
+              .atOffset(ZoneOffset.UTC)
+              .toInstant();
+        }
+      }
+    } catch (FileNotFoundException e) {
+      throw new RuntimeException(e);
```

Comment:
Throw `IllegalArgumentException` instead with a message saying the file is expected to exist. This should never get thrown currently, but that could change in the future. Look at it from the context of this method itself. There's nothing here to ensure the file exists, which means it's up to the caller.